Svenbjorn/whatson
# Fix Caching Bug in Grid Expand Arrow Logic

## Why:
This PR addresses a critical issue in the grid expand arrow feature: expand arrows not deactivating correctly when new grid chunks are loaded. This flaw was identified during dynamic interactions, such as scrolling to add more grid chunks and opening different grid items. To remedy this, we have refactored the expand arrow logic to ensure reliable deactivation of arrows upon grid updates.

The primary objective of this change is to enhance the user experience by ensuring consistent and predictable behavior of expand arrows, particularly in scenarios involving dynamic grid modifications. This update aligns with our commitment to providing a seamless and intuitive interface across our platform.

- Task: [Monday.com](https://visitfaroeislands.monday.com/boards/4732496989/views/112923023/pulses/4533934471)
- Design: [Invision](https://sendistovan.invisionapp.com/console/VFI-Tourism---desktop-ckpjdr19e0hqa010ju83s0gs8/ckpjds4ht0gc0011xa3n7rndz/play)

## Changes Made:
### Caching Mechanism Update:

- Directly targeted the bug of expand arrows not deactivating when new grid chunks are loaded.
- Implemented a new Set-based caching mechanism to track and manage the state of active expand arrows efficiently..

## How to Test:

1. **Grid Amount Change:**

    - Interact with the grid by scrolling down, which adds more chunks to the grid. Open a grid item and then close it. Continue scrolling to add additional grid chunks. Then, open a new grid item and close it. With each action, the grid arrow should consistently close.
    - Perform this test across different sections, including events, places, tours, and hiking, to ensure uniform functionality.

3. **Consistency Check:**

    - Verify that the expand arrows function consistently across the platform, regardless of the grid amount. This includes checking their behavior during and after changes to the grid layout.

4. **Error Handling:**
    - Confirm that there are no errors or unexpected behaviors when the grid amount changes. This includes monitoring for any issues during grid expansion and interaction with the expand arrows.
